server/src/main/java/org/elasticsearch/action/search/SearchRequest.java

```diff
@@ -433,6 +438,25 @@ public void setCcsMinimizeRoundtrips(boolean ccsMinimizeRoundtrips) {
         this.ccsMinimizeRoundtrips = ccsMinimizeRoundtrips;
     }
 
+    /**
+     * Returns whether the "fields" option will be emulated via fetching from source on pre 7.10 nodes.
+     * The default is false.
+     */
+    boolean isFieldsOptionEmulationEnabled() {
+        return enableFieldsEmulation;
+    }
+
+    /**
+     * Sets whether the "fields" option will be emulated via fetching from source on pre 7.10 nodes.
+     * @deprecated This option is only available temporarily. Instead of using this setter directly,
+     * use the parameter on the rest request. The bwc emulation will not be necessary on 8.0 any more,
+     * so this setter will not be availabe there.
+     */
+    @Deprecated
```

Comment:
I added this to warn that this option will be gone in 8.0 already. Unfortunately we need a public setter in the RestSearchAction when parsing the parameter from the request because the classes are in different packages.